internal/proxy/proxy.go

```diff
@@ -61,18 +64,28 @@ type Config struct {
 	Dialer cloudsql.Dialer
 }
 
-// NewConfig initializes a Config struct using the default database engine
-// ports.
-func NewConfig() *Config {
-	return &Config{}
+// Logger defines a minimal interface for logging informational or error
+// messages.
+type Logger interface {
+	// Log reports an informational message.
+	Log(msg string, data ...interface{})
+	// Error reports an error message.
+	Error(msg string, data ...interface{})
 }
 
-func (c *Config) DialerOpts() []cloudsqlconn.Option {
+func (c *Config) DialerOpts(l Logger) []cloudsqlconn.Option {
```

Comment:
Does it make sense to do the logging here? Shouldn't we move it to where the parsing/config is set?

Reply:
Moved this up a level.